hedgehog/src/Hedgehog/Internal/Property.hs

```diff
@@ -152,6 +161,50 @@ newtype PropertyT m a =
     , MonadError e
     )
 
+-- | Classifiers use strings as labels
+--
+type ClassifierName = String
+
+-- | A classifier can be attached to a property conditionally
+--
+--   When the amount of occurrences don't exceed the minimum percentage, a
+--   warning will be issued.
+data Classifier = Classifier
+  { clsMinPercentage :: !Double
+  , clsOccurrences :: !Integer
+  }
+  deriving Show
```

Comment:
I apologise for not having a style guide, but no abbreviations / acronyms where possible (although `x`, `x0`, `x1`, `y` is fine for locals where types are obvious) and also it should be laid out like this:

```haskell
data Classifier =
  Classifier {
      classifierMinimumPercentage :: !Double -- or just classifierPercentage if you prefer
    , classifierOccurrences :: !Integer
    } deriving (Show)
```